### ISSUE TITLE
Trilinos nightly build failure in Ifpack2, "error: no instance of overloaded function "gauss_seidel_numeric" matches the argument list"

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#1327]
Nightly Trilinos builds against kokkos-kernels develop began failing to compile ifpack2 with error message output:

```
[ 71%] Building CXX object packages/ifpack2/src/CMakeFiles/ifpack2.dir/Ifpack2_RILUK_Serial.cpp.o
<https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA101_opt/ws/Trilinos/packages/ifpack2/src/Ifpack2_Relaxation_def.hpp(1334)>: error: no instance of overloaded function "gauss_seidel_numeric" matches the argument list
            argument types are: (KokkosKernels::Experimental::KokkosKernelsHandle<const size_t, int, double, Kokkos::DefaultHostExecutionSpace, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>> *, size_t, size_t, Kokkos::View<const size_t *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::MemoryTraits<0U>>, Kokkos::View<int *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::MemoryTraits<0U>>, Kokkos::View<double *, Kokkos::LayoutRight, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, void>, Kokkos::View<double *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::MemoryTraits<0U>>, __nv_bool)
          detected during instantiation of "void Ifpack2::Relaxation<MatrixType>::compute() [with MatrixType=Tpetra::RowMatrix<double, int, longlong, Kokkos_Compat_KokkosSerialWrapperNode>]"
<https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA101_opt/ws/Build/packages/ifpack2/src/Ifpack2_Relaxation_Serial.cpp(61)>: here
```

The `gauss_seidel_numeric` function referred to in the error message indicates this likely follows merge of PR #1232 

@lucbv @brian-kelley is this a relatively quick fix that should be done for the release, or should #1232 be excluded from 3.6? I'm tentatively labeling as blocks promotion for now

**Reproducer** (weaver cuda build):
kokkos and kokkos-kernels develop branches
Trilinos' kokkos-promotion branch

```
# Environment
export ATDM_CONFIG_REGISTER_CUSTOM_CONFIG_DIR=${TRILINOS_DIR}/cmake/std/atdm/contributed/weaver
source ${TRILINOS_DIR}/cmake/std/atdm/load-env.sh weaver-cuda-10.1-opt
export ATDM_CONFIG_USE_NINJA=OFF
export OMPI_CXX=$KOKKOS_PATH/bin/nvcc_wrapper

# Configure - uses source override to build with kokkos and kokkos-kernels repos
cmake \
 -G"Unix Makefiles" \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DCMAKE_INSTALL_PREFIX="${PWD}/install" \
 -DCMAKE_CXX_STANDARD="14" \
 -DTrilinos_ENABLE_TESTS=ON \
 -DTrilinos_ENABLE_ALL_PACKAGES=ON \
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR
```

This can also be reproduced in a Serial build on Blake

**Comment by brian-kelley** [comment]
Fixed in Trilinos commit d7333823 to kokkos-promotion.

**Comment by ndellingwood** [comment]
Thanks @brian-kelley !
